packages/core/src/agent/Agent.ts

```diff
@@ -156,6 +163,7 @@ export class Agent extends BaseAgent {
     dependencyManager.registerSingleton(TransportService)
     dependencyManager.registerSingleton(Dispatcher)
     dependencyManager.registerSingleton(EnvelopeService)
+    dependencyManager.registerSingleton(FeatureRegistry)
```

Comment:
If feature registry is part of core, I think it should not live in the discover features module.